### PR TITLE
[texi26] run through an example with MOE.

### PR DESCRIPTION
**Pull request by Serina1218**
Summary:
Run through an example with MOE" implements Mixture of Experts (MoE) functionality in the TEXI model architecture. Key changes include:

1.  Configuration changes:

    *   Switched from standard Transformer to MoETransformer by creating a new config file `vtexi_25b_moe.yaml`
    *   Set number of experts to 4
    *   Enabled soft MoE routing with `is_softMoE: True`
2.  Code modifications:

    *   Updated the `moe_transformer.py` to support soft MoE routing by adding parameters:
        *   `is_softMoE` (default: False)
        *   `softMoE_temperature` (default: 0.01)
    *   Modified the sparse MLP configuration to use these parameters instead of hardcoded values
    *   Updated references in BUCK files and Python code to point to the new MoE configuration
3.  Performance optimization:

    *   Modified tensor handling in `module.py` to exclude CPU tensors from the FakeTensor check, preventing unnecessary tensor swapping

The changes enable the model to use a soft Mixture of Experts approach where routing between experts is done with a softmax function rather than token-initiated routing, avoiding triggering of DDDS.
dynamic-MOE vs soft-MOE
https://docs.google.com/document/d/11snR6YrdQCwahyQCP61hWnww8voaZ8qtw7TFN61eqm8/edit?tab=t.0#bookmark=id.vgyjua5vuj6

Test Plan:
```
make -f ai_codesign/nonprod/xi_mtia_bench/texi25b/Makefile check_warmup

```
P1880892716

Rollback Plan:

Differential Revision: D78957047

### CODE REVIEW

**Comment by albanD** [comment]
I guess this was opened by mistake. Closing.
Feel free to open a new PR with relevant description if there was something you needed in pytorch!
